lib/hutch/broker.rb

```diff
@@ -313,6 +321,17 @@ def ensure_connection!(routing_key, message)
       raise_publish_error('connection is closed', routing_key, message) unless @connection.open?
     end
 
+    def channel_broker
+      if Thread.current[CHANNEL_BROKER_KEY] && !Thread.current[CHANNEL_BROKER_KEY].active
+        Thread.current[CHANNEL_BROKER_KEY].reconnect(open_channel!)
+      end
+      Thread.current[CHANNEL_BROKER_KEY] ||= ChannelBroker.new(open_channel!)
```

Comment:
Not convinced that makes it any clearer @Evan, you'd then need to have a final `Thread.current[CHANNEL_BROKER_KEY]` after the if/else statement